homeassistant/components/modbus/sensor.py

```diff
@@ -157,18 +159,20 @@ async def async_setup_platform(
             del entry[CONF_REVERSE_ORDER]
         if entry.get(CONF_SWAP) != CONF_SWAP_NONE:
             if entry[CONF_SWAP] == CONF_SWAP_BYTE:
-                regs_needed = 1
+                size_needed = 2
             else:  # CONF_SWAP_WORD_BYTE, CONF_SWAP_WORD
-                regs_needed = 2
+                size_needed = 4
             if (
-                entry[CONF_COUNT] < regs_needed
-                or (entry[CONF_COUNT] % regs_needed) != 0
+                sizeofbytes < size_needed
+                or (sizeofbytes % size_needed) != 0
+                or sizeofbytes > 8
```

Comment:
that does not make sense? what if I read 20 registers.

Reply:
It overflows > 4 bytes anyway

"OverflowError: int too big to convert"

May need an array or split for swap

I dont know it is fixable

Python3 dont have (int vs long)
But overflow anyway

And you are smart  😉

Pytest pass dont prove that code is working.

My custom fail my tests but works flawlessly. 

Multiple >L
String multiple bytes

I just pushed bad version to you help me

I dont understand yet the code of swap at all

If I fix overflow have struct size issues 

Regards

Let's see what the time will say

Reply:
If I may correct you, you marked that you wanted an interview. Please do not push a bad version. I review your code and I do not have time to do more (not after your comments earlier).

if it overflows, then you should check that the struct is valid (see struct doc),  a long double  is e.g. 16 bytes. and of course advice the user what is wrong.

Your code might work flawlessly, but you need to support all legal configurations. As I have often told you that is a major difference between a personal custom component and HA core code.

The swap code is easy, search in google or look for the swap PR it contains a schema (or the issue it closes has the schema).